common/src/main/java/com/gregtechceu/gtceu/core/MixinHelpers.java

```diff
@@ -92,4 +96,28 @@ public static List<PackResources> addDynamicData(Collection<PackResources> packs
         packResources.add(new GTDynamicDataPack("gtceu:dynamic_data", AddonFinder.getAddons().stream().map(IGTAddon::addonModId).collect(Collectors.toSet())));
         return packResources;
     }
+
+    public static List<PackResources> addDynamicResourcePack(Collection<PackResources> packs) {
+        List<PackResources> packResources = new ArrayList<>(packs);
+        // Clear old data
+        GTDynamicResourcePack.clearClient();
+
+        // Register recipes & unification data again
+        long startTime = System.currentTimeMillis();
+        MaterialBlockRenderer.reinitModels();
+        TagPrefixItemRenderer.reinitModels();
+        ToolItemRenderer.reinitModels();
+        GTModels.registerMaterialFluidModels();
+        GTCEu.LOGGER.info("GregTech Model loading took {}ms", System.currentTimeMillis() - startTime);
+
+        // Load the data
+        packResources.add(new GTDynamicResourcePack("gtceu:dynamic_assets", AddonFinder.getAddons().stream().map(IGTAddon::addonModId).collect(Collectors.toSet())));
+        return packResources;
+    }
+
+    // unused on purpose. Do not call, will destroy ram usage.
+    public static void initializeDynamicTextures() {
```

Comment:
Is this still needed?

Reply:
it might become needed so I left it in for now

Reply:
Alright

Reply:
also it can prove useful for future stuff